internal/component/loki/process/process_test.go

```diff
@@ -164,7 +165,7 @@ func TestJSONLabelsStage(t *testing.T) {
 		"[IN]: timestamp: 2020-11-15T02:08:41-07:00, entry: {\"log\":\"log message\\n\",\"stream\":\"stderr\",\"time\":\"2019-04-30T02:12:41.8443515Z\",\"extra\":\"{\\\"user\\\":\\\"smith\\\"}\"}, labels: {filename=\"/var/log/pods/agent/agent/1.log\", foo=\"bar\"}",
 		"[OUT]: timestamp: 2019-04-30T02:12:41.8443515Z, entry: {\"log\":\"log message\\n\",\"stream\":\"stderr\",\"time\":\"2019-04-30T02:12:41.8443515Z\",\"extra\":\"{\\\"user\\\":\\\"smith\\\"}\"}, labels: {filename=\"/var/log/pods/agent/agent/1.log\", foo=\"bar\", stream=\"stderr\", ts=\"2019-04-30T02:12:41.8443515Z\", user=\"smith\"}",
 	}
-	require.Equal(t, expectedLiveDebuggingLog, liveDebuggingLog.Get())
+	assert.ElementsMatch(t, expectedLiveDebuggingLog, liveDebuggingLog.Get())
```

Comment:
this test is not flaky on main, why would it sometimes not be ordered with your changes?

Reply:
It was failing in my earlier commit https://github.com/grafana/alloy/actions/runs/12394965463/job/34599585193

Reply:
Hey @wildum I missed adding the channel size in the tests have added the same. But I am not sure if that is the reason for failure. I checked the code and logic seems to be good I don't see any race conditions. Like you mentioned the IN and OUT are published in order and should be received in order. I tried to run this in local a couple of times frequently with the channel size fix it seems stable. Can you help me see if I am missing any race condition on this test failure ?

Reply:
ohh, You mean we should remove the config `buffer_stream_size` ? and have only the flush interval changes ? or just remove the config changes for the tests made in this [commit](https://github.com/grafana/alloy/pull/2294/commits/62c8855695a38f6efff6d6d304d8cd66e6f421a8)

Reply:
I think that we should remove the config buffer_stream_size because it doesn't seem necessary in the end